frontend/src/App.test.tsx

```diff
@@ -155,39 +139,48 @@ describe("App", () => {
     expect(instance.connectionManager.disconnect).toHaveBeenCalled()
   })
 
-  it("reloads when streamlit server version changes", () => {
-    const props = getProps()
-    const wrapper = shallow(<App {...props} />)
-
-    // A HACK to mock `window.location.reload`.
-    // NOTE: The mocking must be done after mounting,
-    // but before `handleMessage` is called.
-    const { location } = window
-    // @ts-ignore
-    delete window.location
-    // @ts-ignore
-    window.location = { reload: jest.fn() }
-
-    const fwMessage = new ForwardMsg()
-    fwMessage.newSession = {
-      config: {},
-      initialize: {
-        environmentInfo: {
-          streamlitVersion: "svv",
+  describe("streamlit server version changes", () => {
```

Comment:
This test was refactored to use `beforeEach`/`afterEach` for saving and restoring `window.location`. (In its previous incarnation, a test failure would prevent the window.location from being restored, which would then cascade into many more test failures in unrelated tests.)